frontend/packages/data-portal/app/components/Dataset/DatasetAuthors.tsx

```diff
@@ -39,56 +36,56 @@ export function DatasetAuthors({
       !(author.primary_author_status || author.corresponding_author_status),
   )
 
-  const envelopeIcon = (
-    <EnvelopeIcon className="text-sds-gray-400 mx-sds-xxxs align-top inline-block h-sds-icon-xs w-sds-icon-xs" />
-  )
-
   const otherCollapsed = useMemo<string | null>(() => {
     const ellipsis = '...'
 
     if (compact && authorsOther.length > 0) {
       if (authorsCorresponding.length === 0) {
         return ellipsis
       }
-      return `${ellipsis} ${separator} `
+      return `${ellipsis} ${SEPARATOR}`
     }
     return null
-  }, [authorsOther, authorsCorresponding, compact, separator])
+  }, [authorsCorresponding.length, authorsOther.length, compact])
```

Comment:
nit: I personally don't trust `.length` in the sense that the underlying author objects could change but remain with the same array length. 

Could be an overkill, but maybe a function like `getAuthorIds(authorsCorresponding)` that extract author ids as an array and then run `String()` on it to serialize it, so we get:

```ts
[String(getAuthorIds(authorsCorresponding)), String(getAuthorIds(authorsOther)), compat]
```

Reply:
Awesome! Yeah pretty tricky and annoying lol Also not sure if it'd ever happen given your data, but sometimes I even sort the ids to be SUPER safe 😂